packages/react-dom/src/__tests__/ReactDOMFloat-test.js

```diff
@@ -382,94 +363,127 @@ describe('ReactDOMFloat', () => {
   it('warns if you render resource-like elements above <head> or <body>', async () => {
     const root = ReactDOMClient.createRoot(document);
 
-    renderSafelyAndExpect(
-      root,
-      <>
-        <noscript>foo</noscript>
-        <html>
-          <body>foo</body>
-        </html>
-      </>,
-    ).toErrorDev(
+    await expect(async () => {
+      root.render(
+        <>
+          <noscript>foo</noscript>
+          <html>
+            <body>foo</body>
+          </html>
+        </>,
+      );
+      const aggregateError = await waitForThrow();
+      expect(aggregateError.errors.length).toBe(2);
+      expect(aggregateError.errors[0].message).toContain(
+        'Invalid insertion of NOSCRIPT',
+      );
+      expect(aggregateError.errors[1].message).toContain(
+        'The node to be removed is not a child of this node',
+      );
```

Comment:
I think we can remove the !__DEV__ gate too now

Reply:
Huh I was wondering why the negative flag check wouldn't have caught that and it's because `enableFloat` is `true` in all our builds, so the combined gate condition is always true. Can we just clean up the flag now?

Reply:
Given that this already rolled out at Meta I vote for landing the `enableFloat` flag and removing the associated gates from all our tests. Can do in a follow up.